libs/ledger-live-common/src/families/aptos/getFeesForTransaction.ts

```diff
@@ -78,53 +75,31 @@ export const getFee = async (
           }
         }
       }
-
-      gasLimit =
-        Number(completedTx.gas_used) ||
-        Math.floor(Number(transaction.options.maxGasAmount) / ESTIMATE_GAS_MUL);
+      res.fees = expectedGas;
+      res.estimate.maxGasAmount = gasLimit.toString();
+      res.estimate.gasUnitPrice = completedTx.gas_unit_price;
     } catch (error: any) {
       log(error.message);
       throw error;
     }
   }
-
-  try {
-    const { sequence_number } = await aptosClient.getAccount(account.freshAddress);
-    sequenceNumber = sequence_number;
-  } catch (_) {
-    // skip
-  }
-
-  gasLimit = Math.ceil(gasLimit * ESTIMATE_GAS_MUL);
-
-  res.estimate.gasUnitPrice = gasPrice.toString();
-  res.estimate.sequenceNumber = sequenceNumber.toString();
-  res.estimate.maxGasAmount = gasLimit.toString();
-
-  res.fees = res.fees.plus(BigNumber(gasPrice)).multipliedBy(BigNumber(gasLimit));
-  CACHE.delete(getCacheKey(transaction));
   return res;
 };
 
-const getCacheKey = (transaction: Transaction): string =>
-  JSON.stringify({
-    amount: transaction.amount,
-    gasUnitPrice: transaction.options.gasUnitPrice,
-    maxGasAmount: transaction.options.maxGasAmount,
-    sequenceNumber: transaction.options.sequenceNumber,
-    expirationTimestampSecs: transaction.options.expirationTimestampSecs,
-  });
-
 export const getEstimatedGas = async (
   account: Account,
   transaction: Transaction,
   aptosClient: AptosAPI,
 ): Promise<IGetEstimatedGasReturnType> => {
-  const key = getCacheKey(transaction);
-
-  if (!CACHE.has(key)) {
-    CACHE.set(key, await getFee(account, transaction, aptosClient));
+  if (!CACHE.amount.eq(transaction.amount)) {
```

Comment:
We should use makeLRUCache standard method here (pinpointed it on the main feature branch but not enough time), we can merge your PR as it cleans some parts of code

It's in @ledgerhq/live-network/cache

Reply:
@hedi-edelbloute does makeLRUCache removes the need for the CACHE?

Reply:
yes, makes sense for use to use these CACHE tbh, but I am ok in doing in another iteration